From 06fa63deb7dda686ba26aaff61243a50fcdce2af Mon Sep 17 00:00:00 2001
From: Xisco Fauli <xiscofauli@libreoffice.org>
Date: Mon, 9 Jun 2025 20:25:34 +0200
Subject: [PATCH] UITest_writer_tests8: use wait_until_property_is_updated in
 test_tdf145158

According to https://bugs.documentfoundation.org/show_bug.cgi?id=145158#c20
the issue is still reproducible after
commit eb000710c54fcd6d00735bc74b0ffcc0f54c70e0
Author: Christian Lohmaier <lohmaier+LibreOffice@googlemail.com>
Date:   Sun Jun 8 21:01:47 2025 +0200

    workaround timing issue in UITest_writer_tests8 test_tdf145158

Change-Id: If771cd1ab795b7ea6690ddd8f41c9d98bd13ce50
---
 sw/qa/uitest/writer_tests8/tdf145158.py | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/sw/qa/uitest/writer_tests8/tdf145158.py b/sw/qa/uitest/writer_tests8/tdf145158.py
index ed2d3bd95004..41dc0f193cd1 100644
--- a/sw/qa/uitest/writer_tests8/tdf145158.py
+++ b/sw/qa/uitest/writer_tests8/tdf145158.py
@@ -22,9 +22,7 @@ class tdf145158(UITestCase):
 
             xFontsize = xMainWindow.getChild("fontsizecombobox")
 
-            xToolkit = self.xContext.ServiceManager.createInstance('com.sun.star.awt.Toolkit')
-            xToolkit.waitUntilAllIdlesDispatched()
-
+            self.ui_test.wait_until_property_is_updated(xFontsize, "Text", "12 pt")
             self.assertEqual("12 pt", get_state_as_dict(xFontsize)['Text'])
 
             select_by_text(xFontsize, "10.5 pt")
-- 
2.47.2

